src/api/evaluators/custom_evals/eval_combined.py

```diff
@@ -0,0 +1,333 @@
+import json
+import os
+from datetime import timedelta
+from typing import Any, Dict, List
+from azure.core.exceptions import HttpResponseError
+from azure.identity import DefaultAzureCredential
+from azure.monitor.query import LogsQueryClient, LogsQueryStatus
+from dotenv import load_dotenv
+import prompty
+import prompty.azure
+import logging
+
+# Configure logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+
+# Load environment variables
+load_dotenv()
+
+
+
+# Configuration for the Azure OpenAI endpoint
+model_config = {
+    "azure_endpoint": os.environ["AZURE_OPENAI_ENDPOINT"],
+    "api_version": os.environ["AZURE_OPENAI_API_VERSION"],
+}
+
+
+# Authenticate using DefaultAzureCredential
+credential = DefaultAzureCredential()
+logs_client = LogsQueryClient(credential)
+workspace_id= os.environ["APPINSIGHTS_WORKSPACE_ID"]
+
+
+def extract_app_insights_data( timespan_days: int = 2, take_count: int = 5) -> List[Dict[str, Any]]:
+    """
+    Extracts data from Azure Application Insights for multiple interactions.
+
+    Args:
+        workspace_id (str): The workspace ID to query.
```

Comment:
This currently a global, not an argument. It should probably be an argument, but I don't know all the rules of good python code. :)

Reply:
fixed